src/actions/interface.jl

```diff
@@ -6,5 +6,35 @@ entity based on the action on the given roadway.
 """
 propagate(veh::Entity{S,D,I}, action::A, roadway::R, Δt::Float64) where {S,D,I,A,R} = error("propagate not implemented for Entity{$S, $D, $I}, actions $A, and roadway $R")
 
+propagate(veh::Entity{S,D,I}, state::S, roadway::R, Δt::Float64) where {S,D,I,R} = state
 
-propagate(veh::Entity{S,D,I}, state::S, roadway::R, Δt::Float64) where {S,D,I,R}= state
\ No newline at end of file
+"""
+Mapping from actions to vehicle ids. The main use case is for keeping track of the action history
+in the same way as the scene history in the `simulate!` function.
+
+Initialize as
+
+    ActionMapping(a, id)
+
+where `a` is an action which can be used to propagate entities. `id` is the entity identifier.
+"""
+struct ActionMapping{A,I}
+    action::A
+    id::I
+end
+
+function Base.findfirst(id::I, frame::Frame{ActionMapping{A, I}}) where {A,I}
+    for am_index in 1 : frame.n
+        am = frame.entities[am_index]
+        if am.id == id
+            return am_index
+        end
+    end
+    return nothing
+end
+function Records.id2index(frame::Frame{ActionMapping{A,I}}, id::I) where {A,I}
+    entity_index = findfirst(id, frame)
+    entity_index === nothing && throw(BoundsError(frame, id))
```

Comment:
is that line correct? 
I don't understand the syntax, it should behave as
```julia 
if entity_index === nothing
    throw(BoundsError(frame, id))
end
```
Are the two syntax equivalent? 

I like 
```julia
entity_index === nothing ? throw(BoundsError(frame, if)) : nothing 
```` 
if you are looking for a one liner.